src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -216,21 +253,23 @@ public ResponseEntity<Exam> updateExam(@PathVariable Long courseId, @RequestBody
 
         Exam savedExam = examRepository.save(updatedExam);
 
+        // NOTE: We have to get exercise groups as we need them for re-scheduling
+        Exam examWithExercises = examService.findByIdWithExerciseGroupsAndExercisesElseThrow(savedExam.getId());
+
         // We can't test dates for equality as the dates retrieved from the database lose precision. Also use instant to take timezones into account
         Comparator<ZonedDateTime> comparator = Comparator.comparing(date -> date.truncatedTo(ChronoUnit.SECONDS).toInstant());
         if (comparator.compare(originalExam.getVisibleDate(), updatedExam.getVisibleDate()) != 0
                 || comparator.compare(originalExam.getStartDate(), updatedExam.getStartDate()) != 0) {
-            // get all exercises
-            Exam examWithExercises = examService.findByIdWithExerciseGroupsAndExercisesElseThrow(savedExam.getId());
             // for all programming exercises in the exam, send their ids for scheduling
             examWithExercises.getExerciseGroups().stream().flatMap(group -> group.getExercises().stream()).filter(ProgrammingExercise.class::isInstance).map(Exercise::getId)
                     .forEach(instanceMessageSendService::sendProgrammingExerciseSchedule);
         }
 
-        if (comparator.compare(originalExam.getEndDate(), updatedExam.getEndDate()) != 0) {
-            // get all exercises
-            Exam examWithExercises = examService.findByIdWithExerciseGroupsAndExercisesElseThrow(savedExam.getId());
-            examService.scheduleModelingExercises(examWithExercises);
+        // NOTE: if the end date was changed, we need to update student exams and re-schedule exercises
+        if (comparator.compare(originalExam.getEndDate(), savedExam.getEndDate()) != 0) {
+            // TODO: check if there are any problems with that - i.e. TZ issues, only changing working time in UI, ...
```

Comment:
Did you already check this TODO?

Reply:
Yea looks good, and yea, the working time seems to be reliably calculated by the frontend application.